classes/CartRule.php

```diff
@@ -419,11 +415,19 @@ public static function getCustomerCartRules(
         }
 
         $sql_part1 = '* FROM `' . _DB_PREFIX_ . 'cart_rule` cr
-				LEFT JOIN `' . _DB_PREFIX_ . 'cart_rule_lang` crl ON (cr.`id_cart_rule` = crl.`id_cart_rule` AND crl.`id_lang` = ' . (int) $id_lang . ')';
+			LEFT JOIN `' . _DB_PREFIX_ . 'cart_rule_lang` crl ON (cr.`id_cart_rule` = crl.`id_cart_rule` AND crl.`id_lang` = ' . (int) $id_lang . ')';
+
+        $sql_where = ' WHERE ((cr.`id_customer` = ' . (int) $id_customer . ' OR (cr.`id_customer` = 0 AND (cr.`highlight` = 1 OR cr.`code` = ""))) OR cr.`group_restriction` = 1';
```

Comment:
I'm not sure about `OR cr.`group_restriction` = 1';` I think it has been removed on purpose in #16638

Reply:
You are right, we have updated the branch to keep this change introduced in https://github.com/PrestaShop/PrestaShop/pull/16638